src/Azure.Functions.Cli/Common/Constants.cs

```diff
@@ -55,10 +55,10 @@ internal static class Constants
 
         public static readonly Dictionary<WorkerRuntime, IEnumerable<string>> WorkerRuntimeImages = new Dictionary<WorkerRuntime, IEnumerable<string>>
         {
-            { WorkerRuntime.dotnet, new [] { "mcr.microsoft.com/azure-functions/dotnet", "microsoft/azure-functions-dotnet-core2.0", "mcr.microsoft.com/azure-functions/base", "microsoft/azure-functions-base" } },
-            { WorkerRuntime.node, new [] { "mcr.microsoft.com/azure-functions/node", "microsoft/azure-functions-node8" } },
-            { WorkerRuntime.python, new [] { "mcr.microsoft.com/azure-functions/python", "microsoft/azure-functions-python3.6" } },
-            { WorkerRuntime.powershell, new [] { "mcr.microsoft.com/azure-functions/powershell", "microsoft/azure-functions-powershell" } }
+            { WorkerRuntime.dotnet, new [] { "mcr.microsoft.com/azure-functions/dotnet", "microsoft/azure-functions-dotnet-core2.0", "mcr.microsoft.com/azure-functions/base", "microsoft/azure-functions-base", "DOTNET" } },
```

Comment:
I think you'd want to change the logic here a little bit -- https://github.com/Azure/azure-functions-core-tools/blob/dab8daca606d1a6f398b01b6f29ce8f081c34ec9/src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs#L243

If we just add `DOTNET` etc, and I have a custom image, say `foo:bar-dotnet`, it is going to assume that's a known image which is not true.

Moreover, if I have an image with a weird name `myrepository:not-node` with `FUNCTIONS_WORKER_RUNTIME` as `dotnet`. our logic above may fail with this exception - https://github.com/Azure/azure-functions-core-tools/blob/dab8daca606d1a6f398b01b6f29ce8f081c34ec9/src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs#L269

I would suggest changing that logic to be more like "LinuxFxVersion" startswith `"DOCKER|{knownimage}:"` or equals `"DOCKER|{knownimage}"`, or whatever that can take care of such weird corner cases.